docusaurus/docs/recipes/dropdown.md

```diff
@@ -0,0 +1,427 @@
+
+### What is Downshift 🏎?
+
+> Primitives to build simple, flexible, WAI-ARIA compliant React autocomplete, combobox or select dropdown components.
+
+This is how [Kent C. Dodds](https://kentcdodds.com/) describes his library.
+
+Downshift mainly takes care of three things: **managing state**; **user interactions**; and **accessibility**. The rendering task is all up to the developer. This is possible because it uses both [render prop pattern](https://reactjs.org/docs/render-props.html) and [prop getters concept](https://kentcdodds.com/blog/how-to-give-rendering-control-to-users-with-prop-getters) as its core concept.
+
+You can see its documentation [here](https://github.com/downshift-js/downshift#readme) and read [this blog post](https://kentcdodds.com/blog/introducing-downshift-for-react) for further details.
+
+### How it works?
+
+Downshift provides a [`<Downshift>` component](https://github.com/downshift-js/downshift#downshift-component) responsible for providing select and autocomplete/combobox logic as a render prop. It also provides a [set of hooks](https://github.com/downshift-js/downshift#the-react-hooks-api) holding dedicated logic to each behavior.
+
+## Walk-through
+
+### 1. Install `downshift`
+
+```sh
+npm i downshift --save
+```
+
+### 2. Example
+
+Let's say we need to develop a Dropdown component and we have the following data:
+
+```js
+const items = [
+    {
+        label: "January",
+        value: "january"
+    },
+    {
+        label: "February",
+        value: "february"
+    },
+    {
+        label: "March",
+        value: "march"
+    },
+];
+```
+
+We already have our data. What about now? The following code block is a complete example of how to develop a Dropdown component. Check the example first and then follow the explanation to better understand it.
+
+#### 1. Component
+
+ℹ️ We are using [`useSelect` hook](https://github.com/downshift-js/downshift/tree/master/src/hooks/useSelect) to build our example. You can use the [`<Downshift>` component](https://github.com/downshift-js/downshift#downshift-component) as well to achieve the same result. Check an example [here](https://github.com/downshift-js/downshift#usage).
+
+```js
+// Dropdown.js file
+
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { useSelect } from 'downshift';
+import classNames from 'classnames';
+import DropdownItem from 'path-to-dropdown-item';
+
+import styles from './Dropdown.module.css';
+
+const itemToString = (item) => item ? item.value : '';
+const placeholderLabel = 'Choose your option';
+
+const Dropdown = ({ items, onStateChange, className, ...rest }) => {
+    const {
+        isOpen,
+        selectedItem,
+        highlightedIndex,
+        getMenuProps,
+        getItemProps,
+        getToggleButtonProps,
+    } = useSelect({ itemToString, items, onStateChange });
```

Comment:
An `id` prop must be set when we are server rendering downshift. See [comment#429663734](https://github.com/downshift-js/downshift/issues/602#issuecomment-429663734).